src/state/nfts/createNftsStore.ts

```diff
@@ -0,0 +1,660 @@
+const collectionsPromises: Map<Address | string, Map<string, Promise<void>>> = new Map();
+let prunePromise: { address: Address | string; promise: Promise<void>; lastPruneAt: number } | null = null;
+
+const fetchMultipleCollectionNfts = async (collectionId: string, walletAddress: string, isMigration = false): Promise<NftsQueryData> => {
+  let tokensForCategory: string[] = [];
+
+  if (collectionId === 'showcase') {
+    tokensForCategory = (await getShowcase(walletAddress, isMigration)) || [];
+  } else if (collectionId === 'hidden') {
+    tokensForCategory = (await getHidden(walletAddress, isMigration)) || [];
+  }
+
+  if (isMigration) {
+    tokensForCategory = (await migrateTokens(walletAddress, tokensForCategory)) || [];
+  }
+
+  const tokens = tokensForCategory
+    .map((token: string) => parseUniqueId(token))
+    .filter(p => p.network && p.contractAddress && p.tokenId)
+    .reduce(
+      (acc, curr) => {
+        const network = replaceEthereumWithMainnet(curr.network);
+        if (!network) {
+          return acc;
+        }
+
+        acc[network] ||= [];
+        acc[network].push({
+          contractAddress: curr.contractAddress,
+          tokenId: curr.tokenId,
+        });
+        return acc;
+      },
+      {} as Record<string, { contractAddress: string; tokenId: string }[]>
+    );
+
+  const chainIds = useBackendNetworksStore.getState().getChainsIdByName();
+  const response = await arcClient.getNftsMetadata({ tokens, walletAddress });
+
+  if (!response.getNftsMetadata) {
+    return EMPTY_RETURN_DATA;
+  }
+
+  const nftsByCollection = new Map();
+
+  response.getNftsMetadata.forEach(item => {
+    const { network, contractAddress } = parseUniqueId(item.uniqueId);
+    const collectionId = `${network}_${contractAddress}`.toLowerCase();
+    const uniqueAsset = parseUniqueAsset(item, chainIds);
+
+    const existingCollection = nftsByCollection.get(collectionId);
+    if (existingCollection) {
+      existingCollection.set(item.uniqueId.toLowerCase(), uniqueAsset);
+    } else {
+      const newCollection = new Map();
+      newCollection.set(item.uniqueId.toLowerCase(), uniqueAsset);
+      nftsByCollection.set(collectionId, newCollection);
+    }
+  });
+
+  if (isMigration) {
+    const isReadOnlyWallet = getIsReadOnlyWallet();
+    if (!isReadOnlyWallet) {
+      if (collectionId === 'showcase') {
+        await updateWebShowcase(walletAddress, tokensForCategory, true);
+        useOpenCollectionsStore.getState(walletAddress).setCollectionOpen('showcase', true);
+      } else if (collectionId === 'hidden') {
+        await updateWebHidden(walletAddress, tokensForCategory, true);
+      }
+    }
+  }
+
+  return {
+    collections: new Map(),
+    nftsByCollection,
+    pagination: null,
+  };
+};
+
+const fetchNftData = async (params: NftParams): Promise<NftsQueryData> => {
+  try {
+    const { walletAddress, collectionId, isMigration = false } = params;
```

Comment:
I couldn't find a codepath where `isMigration` is true, is there something I'm missing or is that not a possible condition?

Reply:
it happens when fetching hidden / showcase tokens.

https://github.com/rainbow-me/rainbow/blob/26e170ae223fffdc3f999219186197a03cb2557f/src/hooks/useFetchHiddenTokens.ts#L26
https://github.com/rainbow-me/rainbow/blob/26e170ae223fffdc3f999219186197a03cb2557f/src/hooks/useFetchShowcaseTokens.ts#L29